### DISCUSSION TITLE
[@types/react-dom] enterKeyHint not defined on most elements

### DISCUSSION THREAD

**Discussion by brlodi** [DefinitelyTyped/DefinitelyTyped#70409] [Issues with a @types package]
The [HTML Spec](https://html.spec.whatwg.org/multipage/interaction.html#attr-enterkeyhint) and [React DOM docs](https://react.dev/reference/react-dom/components/common#common-props) both specify the `enterKeyHint` prop (or `enterkeyhint` attribute in plain HTML) being global and available on _all elements_ since it also applies to elements with `contentEditable` set. It is currently only included in the types for `HTMLInputElement`.

Originally I was going to report the issue just for `HTMLTextAreaElement`, which was previously flagged in #43985, but further research seems to indicate every version of React that supports `enterKeyHint` at all should support it for all elements (I believe that's React v17 and up)

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
Good catch. Do you want to file a PR?

**Reply by hkleungai** [reply]
Hi, I have opened a [PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70580) for this. Please help review when you have time.
